python/metatensor-torch/metatensor/torch/atomistic/io.py

```diff
@@ -0,0 +1,145 @@
+    zipf = zipfile.ZipFile(path, "r")
+    required_files = ["positions.npy", "cell.npy", "types.npy", "pbc.npy"]
+    for file in required_files:
+        if file not in zipf.namelist():
+            return False
+    return True
+
+
+def _load_system(path: Union[str, Path]) -> System:
+    with zipfile.ZipFile(path, "r") as zipf:
+        positions = torch.from_numpy(np.load(zipf.open("positions.npy")))
+        cell = torch.from_numpy(np.load(zipf.open("cell.npy")))
+        types = torch.from_numpy(np.load(zipf.open("types.npy")))
+        pbc = torch.from_numpy(np.load(zipf.open("pbc.npy")))
+
+        neighbor_list_options_list = []
+        neighbor_lists = []
+        for nl_idx in range(len(zipf.namelist())):
+            if f"neighbor_lists/{nl_idx}/options.json" in zipf.namelist():
+                nl_options = NeighborListOptions.from_json(
+                    zipf.read(f"neighbor_lists/{nl_idx}/options.json")
+                )
+                neighbor_list_options_list.append(nl_options)
+                numpy_buffer = np.load(zipf.open(f"neighbor_lists/{nl_idx}/data.npy"))
+                tensor_buffer = torch.from_numpy(numpy_buffer)
+                neighbor_list = load_block_buffer(tensor_buffer)
+                neighbor_lists.append(neighbor_list)
+
+        extra_data_dict = {}
+        for data_name in zipf.namelist():
+            if "extra_data/" in data_name:
+                key = os.path.basename(data_name).replace(".npy", "")
+                numpy_buffer = np.load(zipf.open(data_name))
+                tensor_buffer = torch.from_numpy(numpy_buffer)
+                extra_data_dict[key] = load_block_buffer(tensor_buffer)
+
+    system = System(
+        positions=positions,
+        cell=cell,
+        types=types,
+        pbc=pbc,
+    )
+
+    for nl_options, nl in zip(neighbor_list_options_list, neighbor_lists):
+        system.add_neighbor_list(nl_options, nl)
+
+    for key, value in extra_data_dict.items():
+        system.add_data(key, value)
+
+    return system
+
+
+def _is_system(data: torch.ScriptObject) -> bool:
+    if not isinstance(data, torch.ScriptObject):
+        return False
+    try:
+        data.positions
+        data.cell
+        data.types
+        data.pbc
+        return True
+    except AttributeError:
+        return False
+
+
+def _save_system(path: Union[str, Path], system: System) -> None:
+    with zipfile.ZipFile(path, "w") as zipf:
+        for nl_idx, nl_options in enumerate(system.known_neighbor_lists()):
+            zipf.writestr(f"neighbor_lists/{nl_idx}/options.json", nl_options.to_json())
+            nl = system.get_neighbor_list(nl_options)
+            tensor_buffer = metatensor_torch_save_buffer(nl)
+            numpy_buffer = tensor_buffer.numpy()
+            nl_buffer = io.BytesIO()
+            np.save(nl_buffer, numpy_buffer)
+            nl_buffer.seek(0)
+            zipf.writestr(f"neighbor_lists/{nl_idx}/data.npy", nl_buffer.read())
```

Comment:
This creates a npy file containing the bytes of the metatensor npz:

```
$ file neighbor_lists/0/data.npy
neighbor_lists/0/data.npy: NumPy data file, version 1.0, description {'descr': '|u1', 'fortran_order': False, 'shape': (5500,), }

$ unzip neighbor_lists/0/data.npy
Archive:  neighbor_lists/0/data.npy
warning [neighbor_lists/0/data.npy]:  128 extra bytes at beginning or within zipfile
```

Was this intentional? I guess it would make sure the file is a valid npz, at the cost of making reading it harder.

I think I would rather have a format which is not quite NPZ but easier to handle, so using something like this instead:

```suggestion
            tensor_buffer = metatensor_torch_save_buffer(nl)
            zipf.writestr(f"neighbor_lists/{nl_idx}/data.npz", tensor_buffer.numpy().tobytes())
```

We are already saving JSON in there anyway, so it is not going to be a valid npz. But we could maybe change the extension used by all of these, to communicate it more clearly to the users. `.mta` maybe, if it is not already taken? Or `.mts` and also change the default on metatensor -non-atomistic save/load?